### PR TITLE
Dependency check

### PR DESCRIPTION
**Pull request by tevosouza**
@giggio, I think that your project is very useful and I made some changes

**PS**: The document format changes were made by [Bash Beautify](https://marketplace.visualstudio.com/items?itemName=shakram02.bash-beautify) and I can remove them if it can be a problem.

### CODE REVIEW

**Comment by giggio** [comment]
Hey @tevosouza, thanks for suggestions. I don't like the new syntax, I prefer the way the bash file currently is.
And why the change from cmd to entrypoint? Any particular reason?

**Comment by tevosouza** [comment]
No problem I'll remove the beauty bash changes.
The change from CMD to ENTRYPOINT is to allow sending arguments to the docker container

`$ docker run speedtest:latest --verbose` or $ `docker run speedtest:latest --simulate`

If you use CMD as docker startup entry point an error is returned
``` 
$ docker run speedtest:latest --verbose
node: bad option: --verbose ```

File: run.sh [hunk 1]
```diff
@@ -31,9 +31,13 @@ if ! hash jq 2>/dev/null; then
   echo 2>&1 "No 'jq' available."
   exit 1
 fi
-if ! hash speed-test 2>/dev/null; then
```
Comment (giggio) [inline run.sh:34]: You still need this check, the npm tools directory might not be in the PATH.
In this scenario this check fails and `npm list -g` succeeds. In that case, you could manually add it to the PATH before running.

**Review by giggio — CHANGES_REQUESTED** [review]

File: run.sh (hunk 1 above)
Comment (tevosouza) [reply run.sh:34]: Got it,
What do you think about adding the check for npm executable and stop script execution if it not found?
```sh
if ! hash npm 2>/dev/null; then
exit 1
fi 
```

Comment (giggio) [reply run.sh:34]: It is a good addition too.

**Comment by giggio** [comment]
@tevosouza I've moved the project to be Rust based. There is still a branch for the node.js work, if you want to proceed with the PR, but moving forward the new one is the one I'll be using.

**Comment by tevosouza** [comment]
> @tevosouza I've moved the project to be Rust based. There is still a branch for the node.js work, if you want to proceed with the PR, but moving forward the new one is the one I'll be using.

Sadly I never had contact with rust language and I cannot play with it at this time ;( 
I'll close this PR as there was a change of technology and I see no sense in maintaining it using the old one.
Cya! ;)

**Pull request closed without merging**